### DISCUSSION TITLE
Tune XGboost with MLforecast and optuna for time series with multiple unique ids

### DISCUSSION THREAD

**Discussion by JY-2015** [Nixtla/mlforecast#318] [Q&A]
was trying to tune XGboost with MLforecast and optuna for time series with multiple unique ids, any idea how to get the best params for each unique id after optuna study?

**Comment by jmoralez** [comment]
Hey @JY-2015, thanks for using mlforecast. Are you training one model per id? mlforecast trains a single model for all ids

**Reply by JY-2015** [reply]
Thanks for response! correct！ one model per id，each id is supposed to have a set of unique params respectively, then later it can be applied to forecast of each id, not all with one set of parameters.
So any idea what can be used for my case?

**Reply by jmoralez** [reply]
In that case you probably need one study per id, something like the following:
```python
studies = []
for id in train['unique_id'].unique():
    subs = train[train['unique_id'] == id]
    study = optuna.create_study()
    obj = functools.partial(objective, train=subs)
    study.optimize(obj)
    studies.append(study)
```
